plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1123,7 +1122,10 @@ private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecut
                 IcebergSessionProperties.EXPIRE_SNAPSHOTS_MIN_RETENTION);
 
         long expireTimestampMillis = session.getStart().toEpochMilli() - retention.toMillis();
-        expireSnapshots(table, expireTimestampMillis, session, executeHandle.getSchemaTableName());
+
+        table.expireSnapshots()
+                .expireOlderThan(expireTimestampMillis)
+                .commit();
```

Comment:
btw what you did is pretty much the same as what I called few lines below, see 
`table.expireSnapshots().expireOlderThan(expireTimestamp).cleanExpiredFiles(false).commit();` 
See line 1169.